core/server/models/post.js

```diff
@@ -327,6 +327,12 @@ Post = ghostBookshelf.Model.extend({
     onCreating: function onCreating(model, attr, options) {
         options = options || {};
 
+        if (this.get('id') && !ObjectId.isValid(this.get('id'))) {
```

Comment:
There are two things to discuss here:

1. Should it be allowed to create a resource with a custom id via the API? I don't think so.

2. We need this condition for every resource, not just for posts. The check should live in the base model, [here](https://github.com/TryGhost/Ghost/blob/master/core/server/models/base/index.js#L139). The only exception is that the resource id can be 1 for the owner.